### ISSUE TITLE
Spy async bug introduced in Python 3.8.1

### ISSUE THREAD

**Issue by adamcharnock** [pytest-dev/pytest-mock#178]
I believe [bpo-38875](https://bugs.python.org/issue38857) has introduced a bug in `mocker.spy()`. Spied coroutines are no longer awaited as they should be.

This test highlights the issue:

```python
import pytest


class MyClass:
    async def method(self):
        return "hi"


@pytest.mark.asyncio
async def test_spy_async(mocker):
    my_obj = MyClass()
    mocker.spy(my_obj, 'method')

    returned = await my_obj.method()

    # Passes
    assert my_obj.method.called

    # AssertionError: assert <coroutine object MyClass.method at 0x10f2a5d40> == 'hi'
    assert returned == "hi"
```

It looks like `spy()` (specifically, `wrapper()` defined within `spy()`) returns a coroutine and expects `unittest`'s `patch()` do to the awaiting, rather than doing the awaiting itself. This worked until bpo-38875 changed this behaviour. Now `unittest` will no longer await retuned awaitables, rather it returns them.

**Comment by adamcharnock** [comment]
If anyone needs a quick workaround, I'm using something like this:

```python
    if sys.version_info >= (3, 8):
        from unittest.mock import AsyncMock
        redis_pool.grow = AsyncMock(wraps=redis_pool.grow)
    else:
        mocker.spy(redis_pool, "grow")

    assert redis_pool.grow.call_count == 123
```

`AsyncMock` is new in 3.8

**Comment by nicoddemus** [comment]
Hi @adamcharnock,

Thanks for the report and workaround.

Would you consider opening up a PR with this change?

**Comment by k4nar** [comment]
Thank you very much for this fix! Do you know when it will be released? :pray:

**Comment by nicoddemus** [comment]
Thanks for the ping @k4nar! Just released `3.0.0` with this fix. 👍

**Comment by k4nar** [comment]
Awesome, thank you for the responsiveness! Stay safe :) .
